File: bean-registrar/src/main/java/com/example/demo/InMemoryPostRepository.java
package com.example.demo;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class InMemoryPostRepository implements PostRepository {

    private final Map<UUID, Post> posts = new HashMap<>();

    @Override
    public Flux<Post> findAll() {
        return Flux.fromIterable(posts.values());
    }

    @Override
    public Mono<Post> findById(UUID id) {
        return Mono.justOrEmpty(posts.get(id));
    }

    @Override
    public Mono<UUID> save(Post post) {
        UUID id = UUID.randomUUID();
        Post newPost = new Post(id, post.title(), post.content());
        posts.put(id, newPost);
        return Mono.just(id);
    }

    @Override
    public Mono<Long> deleteById(UUID id) {
        posts.remove(id);
        return Mono.just(1L);
    }

    @Override
    public Mono<Long> update(UUID id, Post post) {
        if (posts.containsKey(id)) {
            Post updatedPost = new Post(id, post.title(), post.content());
            posts.put(id, updatedPost);
            return Mono.just(1L);
        } else {
            return Mono.just(0L);
        }
    }
}

File: bean-registrar-kotlin-dsl/src/main/kotlin/com/example/demo/InMemoryPostRepository.kt
package com.example.demo

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flowOf
import java.util.*

class InMemoryPostRepository : PostRepository {
    companion object{
        private val store: MutableMap<UUID, Post> = mutableMapOf()
    }

    override fun findAll(): Flow<Post> {
        return flowOf(*store.values.toTypedArray())
    }

    override suspend fun findById(id: UUID): Post? {
        return store[id]
    }

    override suspend fun save(post: Post): UUID {
        val id = UUID.randomUUID()
        val newPost = Post(id, post.title, post.content)
        store[id] = newPost
        return id
    }

    override suspend fun deleteById(id: UUID): Long {
        store.remove(id)
        return 1L
    }

    override suspend fun initializeSampleData() {
        listOf("Spring 7", "Bean Registrar")
            .forEach {
                this.save(Post(title = it, content = "Content of $it"))
            }
    }

    override suspend fun update(id: UUID, post: Post): Long {
        if (store.containsKey(id)) {
            val updatedPost = Post(id, post.title, post.content)
            store[id] = updatedPost
            return 1L
        } else {
            return 0L
        }
    }
}